### DISCUSSION TITLE
KeepAlive?

### DISCUSSION THREAD

**Discussion by ymmt2005** [connectrpc/connect-go#456] [Q&A] [answered]
grpc-go has a feature called Keepalive that sends HTTP2 Ping frame to detect broken TCP connections.
https://github.com/grpc/grpc-go/blob/master/Documentation/keepalive.md

We've enabled this in our long-running clients. How can we do this with connect-go?

**Comment by akshayjshah** [comment, marked as answer]
The Go standard library's `x/net/http2` does this by default - see https://github.com/golang/go/issues/31643 and https://go-review.googlesource.com/c/net/+/198040.

**Reply by ymmt2005** [reply]
Thank you!
